src/main/kotlin/se/uulm/snowballr/backend/service/ProjectService.kt

```diff
@@ -32,11 +49,32 @@ interface IProjectService {
  * @constructor Initializes the [ProjectService] with a project repository.
  * @param repo The repository responsible for managing persistence operations for projects.
  * @param userRepo The repository responsible for managing persistence operations for users.
+ * @param projectMemberRepo The repository responsible for managing persistence operations for project members.
  */
 class ProjectService(
     private val repo: IProjectTableRepo,
     private val userRepo: IUserTableRepo,
+    private val projectMemberRepo: IProjectMemberTableRepo,
 ) : IProjectService {
+    override suspend fun getProjectById(projectId: Base.Id): ProjectOuterClass.Project {
+        val currentUser = userRepo.getUserById(GrpcContext.getUserIdFromContext())
+        val projectId = parseUUID(projectId.id, EntityType.PROJECT)
+        if (!projectMemberRepo.getMembersOfProject(
+                projectId,
+            ).map { member -> member.userId }.contains(currentUser.id)
+        ) {
```

Comment:
This condition is a bit challenging to read. What about extracting it to another variable and then checking the variable?


```suggestion

        val isInProject = projectMemberRepo.getMembersOfProject(projectId).any { it.userId == currentUser.id }
        if (!isInProject) {
```

Reply:
Here you can use `any` again, instead of `map` and `contains`.